impl/src/main/java/org/glassfish/json/JsonObjectBuilderImpl.java

```diff
@@ -235,9 +235,10 @@ public String getString(String name) {
 
         @Override
         public String getString(String name, String defaultValue) {
-            try {
-                return getString(name);
-            } catch (Exception e) {
+            JsonValue value = get(name);
+            if (value instanceof JsonString) {
```

Comment:
why is this better than `if (value.getValueType() == ValueType.STRING)` ? just asking...

Reply:
I think they are functionally equivalent but your version would also need a null check:

`if (value == null && value.getValueType() == ValueType.STRING)`

I typically avoid instanceof for performance reasons.  But my understanding that this may be old fashioned as the performance of instanceof these days is good.   But I haven't tested it.

Reply:
My understanding is that instanceof is sufficiently fast (as in, it is incredibly unlikely to show up as a hot spot in your profile), but if you would prefer `value != null && value.getValueType() == ValueType.STRING` I would be happy to change to that.

Reply:
makes sense, thanks for summing it up!